use-cases/virgo/data.py

```diff
@@ -61,159 +67,158 @@ def execute(self) -> pd.DataFrame:
-        if dataset is None:
-            print("WARNING: loading time series dataset from disk.")
-            return pd.read_pickle(self.images_dataset)
-        return dataset
+        Args:
+            file_paths (list[str]): List of paths to pickled DataFrames.
+        """
+        self.file_paths = file_paths
 
-    @monitor_exec
-    def execute(
-        self,
-        dataset: Optional[pd.DataFrame] = None
-    ) -> Tuple:
-        """Splits a dataset into train, validation and test splits.
+    def __len__(self):
+        """Return the total number of files in the dataset."""
+        return len(self.file_paths)
+
+    def __getitem__(self, idx):
+        """Retrieve a data sample by index, convert to tensor, and normalize.
 
         Args:
-            dataset (pd.DataFrame): input dataset.
+            idx (int): Index of the file to retrieve.
 
         Returns:
-            Tuple: tuple of train, validation and test splits. Test is None.
+            torch.Tensor: Concatenated and normalized data tensor of main and auxiliary channels.
         """
-        dataset = self.get_or_load(dataset)
+        # Load a single dataframe from the file
+        dataframe = pd.read_pickle(self.file_paths[idx])
 
-        # Convert data to torch
-        df = dataset.applymap(lambda x: torch.tensor(x))
+        # Convert all data in the DataFrame to torch tensors
+        df = dataframe.map(lambda x: torch.tensor(x))
 
-        # Divide Image dataset in main and aux channels. Note that df
-        # generated in the section Generate Synthetic Dataset will always have
-        # the main channel as its first column
+        # Divide Image dataset in main and aux channels.
         main_channel = list(df.columns)[0]
         aux_channels = list(df.columns)[1:]
 
+        # Ensure that there are at least 3 auxiliary channels
+        if len(aux_channels) < 3:
+            print(f"Item with the index {idx} only has {len(aux_channels)} channels!")
+            return None
+
+        # Extract the main channel and auxiliary channels
         df_aux_all_2d = pd.DataFrame(df[aux_channels])
         df_main_all_2d = pd.DataFrame(df[main_channel])
-        X_train_2d, X_test_2d, y_train_2d, y_test_2d = train_test_split(
-            df_aux_all_2d, df_main_all_2d,
-            test_size=self.validation_proportion, random_state=self.rnd_seed)
-        return (X_train_2d, y_train_2d), (X_test_2d, y_test_2d), None
 
+        # Stack the main and auxiliary channels into 2D tensors
+        signal_data_train_2d = torch.stack(
+            [torch.stack(
+                [df_main_all_2d[main_channel].iloc[i]]) for i in range(df_main_all_2d.shape[0])])
+
+        aux_data_train_2d = torch.stack(
+            [torch.stack(
+                [df_aux_all_2d.iloc[i, 0],
+                    df_aux_all_2d.iloc[i, 1],
+                    df_aux_all_2d.iloc[i, 2]]
+            ) for i in range(df_aux_all_2d.shape[0])]
+        )
```

Comment:
I think in these cases a for loop might be more readable, but I suppose you should consider whether that hurts performance too much. Right now the formatting makes this quite hard to read, at least.